app-backend/batch/src/main/scala/com/azavea/rf/batch/ingest/spark/Ingest.scala

```diff
@@ -221,74 +221,39 @@ object Ingest extends SparkJob with LazyLogging with Config {
         math.max(params.partitionsPerFile, getSizeFromURI(s.uri, s3Client) / (params.partitionsSize * 1024 * 1024))
       }.sum.toInt
 
-    // Read source tiles and reproject them to desired CRS
-    val sourceTiles: RDD[((ProjectedExtent, Int), Tile)] =
-      sc.parallelize(layer.sources, layer.sources.length)
-        .flatMap ({ source =>
-          val geotiff = MultibandGeoTiff(
-            byteReader = uriRangReader(source.uri),
-            decompress = false,
-            streaming = true
-          )
-
-          gridBoundChips(geotiff.tile.gridBounds, params.windowSize, params.windowSize)
-            .map { chipBounds => (source, geotiff.rasterExtent.extentFor(chipBounds)) }
-        })
-        .repartition(repartitionSize)
-        .flatMap { case (source, chipExtent) =>
-          val geotiff = MultibandGeoTiff(
-            byteReader = uriRangReader(source.uri),
-            e = Some(chipExtent)
-          )
-
-          val chip = geotiff.tile
-
-          // Set NoData values if a pattern has been specified
-          val maskedChip = ndPattern.fold(chip)(mask => mask(chip))
-
-          source.bandMaps.map { bm: BandMapping =>
-            // GeoTrellis multi-band tiles are 0 indexed
-            val band = maskedChip.band(bm.source - 1).reproject(chipExtent, geotiff.crs, destCRS)
-            (ProjectedExtent(band.extent, destCRS), bm.target.index - 1) -> band.tile
-          }
-        }
+    val (maxZoom, layerMeta): (Int, TileLayerMetadata[SpatialKey]) =
+      Ingest.calculateTileLayerMetadata(layer, layoutScheme)
 
-    val (maxZoom, tileLayerMetadata) = Ingest.calculateTileLayerMetadata(layer, layoutScheme)
-
-    val tiledRdd = sourceTiles.tileToLayout[(SpatialKey, Int)](
-      tileLayerMetadata.cellType,
-      tileLayerMetadata.layout,
-      resampleMethod)
-
-    // Merge Tiles into MultibandTile and fill in bands that aren't listed
-    val multibandTiledRdd: RDD[(SpatialKey, MultibandTile)] = tiledRdd
-      .map { case ((key, band), tile) => key -> (tile, band) }
-      .groupByKey
-      .map { case (key, tiles) =>
-        val prototype: Tile = tiles.head._1
-        val emptyTile: Tile = ArrayTile.empty(prototype.cellType, prototype.cols, prototype.rows)
-        val arr = tiles.toArray
-        val bands: Seq[Tile] =
-          for (band <- 0 until bandCount) yield
-            arr.find(_._2 == band).map(_._1).getOrElse(emptyTile)
-        key -> MultibandTile(bands)
-      }
+    val rawMultis: Array[RDD[(ProjectedExtent, MultibandTile)]] = layer.sources.map { source =>
+      val uri: AmazonS3URI = new AmazonS3URI(source.uri)
+
+      S3GeoTiffRDD.multiband[ProjectedExtent](uri.getBucket, uri.getKey, S3GeoTiffRDD.Options.DEFAULT)
```

Comment:
TODO: Default options are almost certainly not what we need here. `SourceDefinition` has a lot of relevant looking information.